src/rewrite_gelu.cpp

```diff
@@ -53,7 +58,60 @@ struct find_gelu_erf
     }
 };
 
-void rewrite_gelu::apply(module& m) const { match::find_matches(m, find_gelu_erf{}); }
+/**
+ * Find fastGELU blocks (where the graph already does a GELU approximation) and replace them
+ * with an alternative approximation that is less likely to overflow.
+ * The replacement approximation is equivalent to:
+ * GELU(x) ~= 0.5 * x * ( 1 + tanh( sqrt(2/M_PI) * (x + 0.044715 * x^3)))
+ */
+struct find_tanh_fast_gelu
+{
+    auto matcher() const { return match::gelu_tanh(); }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        // Algebraically rearranged version of the tanh GELU approximation that is more numerically
+        // stable
+        auto ins        = r.result;
+        auto x          = r.instructions["x"];
+        auto sqrt_2_rpi = m.add_literal(
+            literal{shape{x->get_shape().type()},
+                    {0.7978845608028653558798921198687637369517172623298693153318516593}});
```

Comment:
Can we write this as `sqrt(2.0f) / M_PI` for the value?

Reply:
Sure, probably want to do the calculation with doubles.